yarn-project/noir-contracts/src/contracts/inclusion_proofs_contract/src/main.nr

```diff
@@ -0,0 +1,285 @@
+    use dep::aztec::{
+        state_vars::{
+            map::Map,
+            set::Set,
+            public_state::PublicState,
+        },
+        selector::compute_selector,
+        types::{
+            address::AztecAddress,
+            type_serialization::field_serialization::FieldSerializationMethods,
+        },
+        context::Context,
+        note::{
+            note_getter_options::NoteGetterOptions,
+            note_header::NoteHeader,
+            utils as note_utils,
+        },
+        constants_gen::{
+            NOTE_HASH_TREE_HEIGHT,
+            NULLIFIER_TREE_HEIGHT,
+            HISTORIC_BLOCKS_TREE_HEIGHT,
+            PUBLIC_DATA_TREE_HEIGHT,
+            GENERATOR_INDEX__PUBLIC_LEAF_INDEX,
+        },
+        oracle::{
+            get_block_data::get_block_data,
+            get_membership_witness::{
+                get_membership_witness,
+                MembershipWitness,
+            },
+            get_sibling_path::get_sibling_path,
+            get_nullifier_membership_witness::{
+                get_low_nullifier_membership_witness,
+                get_nullifier_membership_witness,
+                NullifierMembershipWitness,
+            },
+        },
+        hash::pedersen_hash,
+    };
+    use dep::value_note::value_note::{ValueNote, ValueNoteMethods, VALUE_NOTE_LEN};
+
+    use crate::utils::{
+        full_field_less_than,
+        full_field_greater_than,
+    };
+
+    struct Storage {
+        private_values: Map<Set<ValueNote, VALUE_NOTE_LEN>>,
+        public_value: PublicState<Field, 1>,
+    }
+
+    impl Storage {
+        fn init(context: Context) -> Self {
+            Storage {
+                private_values: Map::new(
+                    context,
+                    1, // Storage slot
+                    |context, slot| {
+                        Set::new(context, slot, ValueNoteMethods)
+                    },
+                ),
+                public_value: PublicState::new(
+                    context,
+                    2, // Storage slot
+                    FieldSerializationMethods,
+                ),
+            }
+        }
+    }
+
+    #[aztec(private)]
+    fn constructor(public_value: Field) {
+        let selector = compute_selector("_initialize(Field)");
+        context.call_public_function(context.this_address(), selector, [public_value]);
+    }
+
+    #[aztec(public)]
+    internal fn _initialize(value: Field) {
+        storage.public_value.write(value);
+    }
+
+    // Creates a value note owned by `owner`.
+    #[aztec(private)]
+    fn create_note(
+        owner: AztecAddress,
+        value: Field,
+    ) {
+        let owner_private_values = storage.private_values.at(owner.address);
+        let mut note = ValueNote::new(value, owner.address);
+        owner_private_values.insert(&mut note, true);
+    }
+
+
+    // Proves that the owner owned a ValueNote at block `block_number`.
+    #[aztec(private)]
+    fn proveNoteInclusion(
```

Comment:
What do you all think about moving this entire fn to `aztec-nr/aztec`. Function would take a note as a parameter along with the existing parameters

Reply:
As I mentioned on slack will do this in a separate PR 😊